custom_components/hacs/api/hacs_repository_data.py

```diff
@@ -35,8 +36,13 @@ async def hacs_repository_data(hass, connection, msg):
         return
 
     if action == "add":
-        if "github." in repo_id:
-            repo_id = repo_id.split("github.com/")[1]
+        # Extract username/repo, stripping URL and trailing characters
+        pattern = r"(?:(?:.*github.com.)|^)([A-Za-z0-9-]+\/[\w.-]+?)(?:(?:\.git)?|(?:[^\w.-].*)?)$"
```

Comment:
@ludeeus good call on the `.git` extensions, forgot about that.

I think this should catch all of our cases. In addition to the previous tests, it also works for:

https://github.com/user/repo.git
git@github.com:user/repo.git

Feel free to clone it and test it out 👍